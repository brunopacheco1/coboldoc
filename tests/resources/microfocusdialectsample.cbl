       >>SOURCE FORMAT FREE
*>>**
*>>  <summary>Short sample.</summary>
*>>  <author>Bruno Pacheco (https://brunopacheco1.github.io/)</author>
*>>  <license>LGPL-3.0</license>
*>>**

*>>*
*>>  <summary>The first module.  
*>>  Trying to see **what** happens to    huge text.</summary>
*>>*
 IDENTIFICATION DIVISION.
 PROGRAM-ID. first-module.

 ENVIRONMENT DIVISION.
 CONFIGURATION SECTION.
 REPOSITORY.
     FUNCTION firstmodulefunction.

 DATA DIVISION.
 WORKING-STORAGE SECTION.
 01 WS-FIRSTMODULE PIC 9.
 01 WS-TRANSFER    PIC 9.
 
 PROCEDURE DIVISION.
 
 0001-FIRSTMODULE-MAIN.
    PERFORM 0002-FIRSTMODULE-NEWPARA.
    STOP RUN.
 
 0002-FIRSTMODULE-NEWPARA.
    ACCEPT WS-FIRSTMODULE.
    COMPUTE WS-FIRSTMODULE = firstmodulefunction(WS-FIRSTMODULE).
    DISPLAY "First Module: ", WS-FIRSTMODULE.
    CALL 'second-module' USING BY CONTENT WS-FIRSTMODULE
    END-CALL.
 END PROGRAM first-module.
 
*>>*
*>> <summary>The second module</summary>
*>>*
 IDENTIFICATION DIVISION.
 PROGRAM-ID. second-module.

 ENVIRONMENT DIVISION.
 CONFIGURATION SECTION.
 REPOSITORY.
     FUNCTION secondmodulefunction.

 DATA DIVISION.
 WORKING-STORAGE SECTION.
 01 WS-SECONDMODULE PIC 9.
   
 LINKAGE SECTION.
 01 WS-TRANSFER    PIC 9.
 
 PROCEDURE DIVISION USING WS-TRANSFER.

 0001-SECONDMODULE-MAIN.
    PERFORM 0002-SECONDMODULE-NEWPARAGRAPH.
    STOP RUN.

 0002-SECONDMODULE-NEWPARAGRAPH.
    MOVE WS-TRANSFER TO WS-SECONDMODULE.
    COMPUTE WS-SECONDMODULE = secondmodulefunction(WS-SECONDMODULE).
    DISPLAY "Second Module: ", WS-SECONDMODULE.
    CALL 'third-module' USING BY CONTENT WS-SECONDMODULE
    END-CALL.

 END PROGRAM second-module.
 
*>>*
*>> <summary>The third module</summary>
*>>*
 IDENTIFICATION DIVISION.
 PROGRAM-ID. third-module.

 ENVIRONMENT DIVISION.
 CONFIGURATION SECTION.
 REPOSITORY.
     FUNCTION thirdmodulefunction.

 DATA DIVISION.
 WORKING-STORAGE SECTION.
 01 WS-THIRDMODULE PIC 9.
   
 LINKAGE SECTION.
 01 WS-TRANSFER    PIC 9.
 
 PROCEDURE DIVISION USING WS-TRANSFER.

 0001-THIRDMODULE-MAIN.
    PERFORM 0002-THIRDMODULE-NEWPARAGRAPH.
    STOP RUN.

 0002-THIRDMODULE-NEWPARAGRAPH.
    MOVE WS-TRANSFER TO WS-THIRDMODULE.
    COMPUTE WS-THIRDMODULE = thirdmodulefunction(WS-THIRDMODULE).
    DISPLAY "Third Module: ", WS-THIRDMODULE.
 END PROGRAM third-module.

*>>*
*>> <summary>first module function</summary>
*>> <param type="PIC 9" name="first-arg">First arg</param>
*>> <param type="PIC 9" name="second-arg">Second arg</param>
*>> <returns type="PIC 9">First return</returns>
*>>*
IDENTIFICATION DIVISION.
FUNCTION-ID. firstmodulefunction.

DATA DIVISION.
WORKING-STORAGE SECTION.
LINKAGE SECTION.
   01 first-arg PIC 9.
   01 firstresult PIC 9.
PROCEDURE DIVISION USING first-arg RETURNING firstresult.
    COMPUTE firstresult = firstarg + 1.
END FUNCTION firstmodulefunction.

*>>*
*>> <summary>second module function</summary>
*>> <param type="PIC 9" name="secondarg">Second arg</param>
*>> <returns type="PIC 9">Second return</returns>
*>>*
IDENTIFICATION DIVISION.
FUNCTION-ID. secondmodulefunction.

DATA DIVISION.
WORKING-STORAGE SECTION.
LINKAGE SECTION.
   01 secondarg PIC 9.
   01 secondresult PIC 9.
PROCEDURE DIVISION USING secondarg RETURNING secondresult.
    COMPUTE secondresult = secondarg + 2.
END FUNCTION secondmodulefunction.

*>>*
*>> <summary>third module function</summary>
*>> <param type="PIC 9" name="thirdarg">Third arg</param>
*>> <returns type="PIC 9">Third return</returns>
*>>*
IDENTIFICATION DIVISION.
FUNCTION-ID. thirdmodulefunction.

DATA DIVISION.
WORKING-STORAGE SECTION.
LINKAGE SECTION.
   01 thirdarg PIC 9.
   01 thirdresult PIC 9.
PROCEDURE DIVISION USING thirdarg RETURNING thirdresult.
    COMPUTE thirdresult = thirdarg + 3.
END FUNCTION thirdmodulefunction.
