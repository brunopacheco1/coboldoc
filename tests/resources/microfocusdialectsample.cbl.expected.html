<!DOCTYPE>
<html lang="en" xml:lang="en">

<head>
    <title>microfocusdialectsample.cbl</title>
    <meta content="text/html; charset=utf-8" />
    <style type="text/css">
        table {
            border-collapse: collapse;
        }

        th {
            background: #dee3e9;
            padding: 7px;
        }

        td {
            padding: 7px;
        }

        tr:nth-child(even) {
            background: #FFF
        }

        tr:nth-child(odd) {
            background: #CCC
        }
    </style>
</head>

<body>
    <h1>
        microfocusdialectsample.cbl
    </h1>
    <p>
        Short sample.
    </p>
    <p>
        author: Bruno Pacheco (https://brunopacheco1.github.io/)
    </p>
    <p>
        License: LGPL-3.0
    </p>
    <h2>Module Summary</h2>
    <table>
        <caption>Module Summary</caption>
        <tr>
            <th scope="col">Name</th>
            <th scope="col">Description</th>
        </tr>
        <tr>
            <td><a href="#first-module">first-module</a></td>
            <td>The first module.  
Trying to see **what** happens to    huge text.</td>
        </tr>
        <tr>
            <td><a href="#second-module">second-module</a></td>
            <td>The second module</td>
        </tr>
        <tr>
            <td><a href="#third-module">third-module</a></td>
            <td>The third module</td>
        </tr>
    </table>
    <h2>Function Summary</h2>
    <table>
        <caption>Function Summary</caption>
        <tr>
            <th scope="col">Name</th>
            <th scope="col">Description</th>
        </tr>
        <tr>
            <td><a href="#firstmodulefunction">firstmodulefunction</a></td>
            <td>first module function</td>
        </tr>
        <tr>
            <td><a href="#secondmodulefunction">secondmodulefunction</a></td>
            <td>second module function</td>
        </tr>
        <tr>
            <td><a href="#thirdmodulefunction">thirdmodulefunction</a></td>
            <td>third module function</td>
        </tr>
    </table>
    <h2>Module Details</h2>
    <h3 id="first-module">first-module</h3>
    <p>
        <em>CALL "first-module".</em>
    </p>
    <p>The first module.  
Trying to see **what** happens to    huge text.</p>
    
    <h3 id="second-module">second-module</h3>
    <p>
        <em>CALL "second-module".</em>
    </p>
    <p>The second module</p>
    
    <h3 id="third-module">third-module</h3>
    <p>
        <em>CALL "third-module".</em>
    </p>
    <p>The third module</p>
    
    <h2>Function Details</h2>
    <h3 id="firstmodulefunction">firstmodulefunction</h3>
    <p>
        <em>{PIC 9} firstmodulefunction(first-arg, second-arg)</em>
    </p>
    <p>first module function</p>
    <h4>Parameters</h4>
    <ul>
        <li>
            <em>{PIC 9}</em> <strong>first-arg</strong> First arg
        </li>
        <li>
            <em>{PIC 9}</em> <strong>second-arg</strong> Second arg
        </li>
    </ul>
    <h4>Returns</h4>
    <ul>
        <li>
            <em>{PIC 9}</em> First return
        </li>
    </ul>
    <h3 id="secondmodulefunction">secondmodulefunction</h3>
    <p>
        <em>{PIC 9} secondmodulefunction(secondarg)</em>
    </p>
    <p>second module function</p>
    <h4>Parameters</h4>
    <ul>
        <li>
            <em>{PIC 9}</em> <strong>secondarg</strong> Second arg
        </li>
    </ul>
    <h4>Returns</h4>
    <ul>
        <li>
            <em>{PIC 9}</em> Second return
        </li>
    </ul>
    <h3 id="thirdmodulefunction">thirdmodulefunction</h3>
    <p>
        <em>{PIC 9} thirdmodulefunction(thirdarg)</em>
    </p>
    <p>third module function</p>
    <h4>Parameters</h4>
    <ul>
        <li>
            <em>{PIC 9}</em> <strong>thirdarg</strong> Third arg
        </li>
    </ul>
    <h4>Returns</h4>
    <ul>
        <li>
            <em>{PIC 9}</em> Third return
        </li>
    </ul>
</body>

</html>